### PR TITLE
[FLINK-27822][docs-zh] Translate the doc of checkpoint/savepoint guar…

### PR DESCRIPTION
**Pull request by zoltar9264**
…antees

## What is the purpose of the change

Translate the doc of checkpoint/savepoint guarantees to Chinese, described in [FLINK-27822](https://issues.apache.org/jira/browse/FLINK-27822) .

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md [hunk 1]
```diff
@@ -24,72 +24,65 @@ specific language governing permissions and limitations
...
-- Checkpoints are stored in state backend-specific (native) data format (may be incremental depending on the specific backend).
+- 在用户终止作业后，会自动删除 Checkpoint（除非明确配置为保留的 Checkpoint）。
+- Checkpoint 以状态后端特定的（原生的）数据格式存储（根据特定后端可能是增量的）。
 {{< /hint >}}
 
-Although [savepoints]({{< ref "docs/ops/state/savepoints" >}}) are created internally with the same mechanisms as
-checkpoints, they are conceptually different and can be a bit more expensive to produce and restore from. Their design focuses
-more on portability and operational flexibility, especially with respect to changes to the job. 
-The use case for savepoints is for planned, manual operations. For example, this could be an update of your Flink version, changing your job graph, and so on.
+尽管 [savepoints]({{< ref "docs/ops/state/savepoints" >}}) 在内部使用与 checkpoints 相同的机制创建，但它们在概念上有所不同，并且生成和恢复的成本可能会更高一些。他们的设计更侧重于可移植性和操作灵活性，尤其是在 job 变更方面。Savepoint 的用例是计划中的、手动的运维。例如，可能是你的 Flink 版本更新，更改你的作业图等等。
```
Comment (JingGe) [inline docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:47]:
```suggestion
尽管 [savepoints]({{< ref "docs/ops/state/savepoints" >}}) 在内部使用与 checkpoints 相同的机制创建，但它们在概念上有所不同，并且生成和恢复的成本可能会更高一些。Savepoints的设计更侧重于可移植性和操作灵活性，尤其是在 job 变更方面。Savepoint 的用例是针对计划中的、手动的运维。例如，可能是更新你的 Flink 版本，更改你的作业图等等。
```

**Review by JingGe — COMMENTED** [review]
Thanks @zoltar9264 for your contribution. It looks overall really good. I just left one comment.

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md [hunk 2]
```diff
@@ -24,72 +24,65 @@ specific language governing permissions and limitations
...
-i) being as lightweight to create and ii) being as fast to restore from as possible. 
-Optimizations towards those goals can exploit certain properties, e.g., that the job code doesn't change between the execution attempts. 
+## 概述
+
+从概念上讲，Flink 的 [savepoints]({{< ref "docs/ops/state/savepoints" >}}) 与 [checkpoints]({{< ref "docs/ops/state/checkpoints" >}}) 的不同之处类似于传统数据库系统中的备份与恢复日志之间的差异。
+
+Checkpoints 的主要目的是为意外失败的作业提供恢复机制。
+[Checkpoint 的生命周期]({{< ref "docs/dev/datastream/fault-tolerance/checkpointing" >}}) 由 Flink 管理，
+即 Flink 创建，管理和删除 checkpoint - 无需用户交互。
+作为一种恢复和定期触发的方法，Checkpoint 实现有两个设计目标：i）轻量级创建和 ii）尽可能快地恢复。 可能会利用某些特定的属性来达到这个，例如， 工作代码在执行尝试之间不会改变。
```
Comment (1996fanrui) [inline docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:40]:
```suggestion
由于 checkpoint 被经常触发，且被用于作业恢复，所以 Checkpoint 的实现有两个设计目标：i）轻量级创建和 ii）尽可能快地恢复。 可能会利用某些特定的属性来达到这个目标，例如， 作业的代码在执行尝试时不会改变。
```

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md [hunk 3]
```diff
@@ -24,72 +24,65 @@ specific language governing permissions and limitations
...
 
 {{< hint info >}}
-- Checkpoints are automatically deleted if the application is terminated by the user 
-(except if checkpoints are explicitly configured to be retained).
-- Checkpoints are stored in state backend-specific (native) data format (may be incremental depending on the specific backend).
+- 在用户终止作业后，会自动删除 Checkpoint（除非明确配置为保留的 Checkpoint）。
+- Checkpoint 以状态后端特定的（原生的）数据格式存储（根据特定后端可能是增量的）。
```
Comment (1996fanrui) [inline docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:44]:
```suggestion
- Checkpoint 以状态后端特定的（原生的）数据格式存储（有些状态后端可能是增量的）。
```

**Review by 1996fanrui — CHANGES_REQUESTED** [review]
Hi @zoltar9264 , thanks for your contribution. I left some comments, please help address in your free time, thanks.

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md (hunk 1 above)
Comment (zoltar9264) [reply docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:47]: Thanks @JingGe , much clear with your suggestion ! committed it.

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md (hunk 3 above)
Comment (zoltar9264) [reply docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:44]: Thinks @1996fanrui , committed it.

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md (hunk 2 above)
Comment (zoltar9264) [reply docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md]: Thanks @1996fanrui , your suggestion really better, committed it.

**Review by JingGe — APPROVED** [review]
LGTM

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md [hunk 4]
```diff
@@ -24,72 +24,65 @@ specific language governing permissions and limitations
...
+| State Processor API (写)          | ✓                   | x                | x                  | x                    |
+| State Processor API (读)          | ✓                   | !                | !                  | x                    |
+| 自包含和可移动                    | ✓                   | ✓                | x                  | x                    |
+| Schema 变更                       | ✓                   | !                | !                  | !                    |
+| 任意 job 升级                     | ✓                   | ✓                | ✓                  | x                    |
+| 非任意 job 升级                   | ✓                   | ✓                | ✓                  | x                    |
+| Flink 小版本升级                  | ✓                   | ✓                | ✓                  | x                    |
+| Flink bug/patch 版本升级          | ✓                   | ✓                | ✓                  | ✓                    |
+| 伸缩                              | ✓                   | ✓                | ✓                  | ✓                    |
```
Comment (Myasuka) [inline docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:74]: `Rescaling`感觉翻译成`扩缩容`是不是更好理解一些？

**Review by Myasuka — COMMENTED** [review]
I think the PR looks good in general. Please take a look at my comments.

File: docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md (hunk 4 above)
Comment (zoltar9264) [reply docs/content.zh/docs/ops/state/checkpoints_vs_savepoints.md:74]: 确实，扩缩容是更习惯的说法，我改下。

**Review by Myasuka — APPROVED** [review]
LGTM, @zoltar9264 could you please squash all commits into one?

**Comment by zoltar9264** [comment]
> LGTM, @zoltar9264 could you please squash all commits into one?

sure , done.

**Comment by Myasuka** [comment]
@flinkbot run azure

**Comment by zoltar9264** [comment]
@flinkbot run azure

**Comment by zoltar9264** [comment]
@flinkbot run azure

**Comment by zoltar9264** [comment]
@flinkbot run azure

**Comment by zoltar9264** [comment]
Hi @Myasuka , It seems that test can't be triggered, have you ever come across this ?

**Comment by Myasuka** [comment]
> Hi @Myasuka , It seems that test can't be triggered, have you ever come across this ?

No, @zentol do you know what happened here?

**Comment by zoltar9264** [comment]
@flinkbot run azure

**Comment by 1996fanrui** [comment]
Hi @zoltar9264 , could you try submit a new PR? I see CI for other PRs is normal.

**Comment by zoltar9264** [comment]
Hi @1996fanrui , I tried to create a new PR , but github doesn't allow it because there is already a PR for the same.

**Comment by Myasuka** [comment]
@zoltar9264 I don't think Github could block submitting a new PR. We use JIRA's issue system to track instead of Github's issue system, I don't know why Github could block this.

@flinkbot run azure

**Comment by zoltar9264** [comment]
Hi @Myasuka , When I try to submit a new PR, github shows the previous PR (this PR), and I can't find submit button.
image

**Comment by Myasuka** [comment]
@zoltar9264 I think this is because you used the same remote forked branch. You might need to create another remote branch to submit another PR.

**Comment by zoltar9264** [comment]
Thanks @Myasuka , exactly as you said. I submit a new [PR](https://github.com/apache/flink/pull/20148) .
@1996fanrui

**Pull request closed without merging**